tools/spy-detector/src/main.rs

```diff
@@ -0,0 +1,262 @@
+use dashmap::DashMap;
+use nng::options::protocol::pubsub::Subscribe;
+use nng::options::Options;
+use nng::{Protocol, Socket};
+use shared::clap;
+use shared::clap::Parser;
+use shared::event_msg;
+use shared::event_msg::event_msg::Event;
+use shared::net_msg;
+use shared::prost::Message;
+use shared::simple_logger;
+use std::sync::atomic;
+use std::sync::atomic::AtomicU32;
+use std::sync::Arc;
+use std::time::Duration;
+use std::time::Instant;
+
+const ADDRESS: &str = "tcp://127.0.0.1:8883";
+const STATS_INTERVAL: Duration = Duration::from_secs(10); // Duration(seconds) to print stats of peers
+
+const LOG_TARGET: &str = "main";
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about=None)]
+struct Args {
+    /// set the threshold for spy detection (default value is 5)
```

Comment:
It would be good to explain what the threshold affects. 

And explain: When would we want to increase or lower it?

Reply:
I just saw that it's not used. Feel free to remove it

Reply:
Threshold is the ratio set by the user to indicate spy nodes; we will need this for detecting spies.

Reply:
Yes, I mean as a comment in the code.

Reply:
Okay, done!